src/ensembl/production/metadata/api/adaptors/genome.py

```diff
@@ -634,27 +636,53 @@ def fetch_genome_datasets(self,
-                                               datasets=[
-                                                   GenomeDatasetItem(
-                                                       dataset=gd.dataset,
-                                                       dataset_type=gd.dataset.dataset_type,
-                                                       dataset_source=gd.dataset.dataset_source,
-                                                       release=gd.dataset.genome_datasets[0],  # I've got the feeling
-                                                       # that this [0] will give birth to a bug in the future... we
-                                                       # will see, it works for now because we have one genome_dataset
-                                                       # linked to a dataset, but we can have many in the future
-                                                       # we can deal with this later when the need arises
-                                                       attributes=[
-                                                           DatasetAttributeItem(name=ds.attribute.name, value=ds.value,
-                                                                                type=ds.attribute.type,
-                                                                                label=ds.attribute.label) for ds in
-                                                           gd.dataset.dataset_attributes]
-                                                   ) for gd in genome_datasets
-                                               ]))
+                    datasets_list = []
+                    for gd in genome_datasets:
+                        # Build attributes first
+                        attributes = [
+                            DatasetAttributeItem(
+                                name=ds.attribute.name,
+                                value=ds.value,
+                                type=ds.attribute.type,
+                                label=ds.attribute.label
+                            )
+                            for ds in gd.dataset.dataset_attributes
+                        ]
+
+                        # Build dataset item
+                        dataset_item = GenomeDatasetItem(
+                            dataset=gd.dataset,
+                            dataset_type=gd.dataset.dataset_type,
+                            dataset_source=gd.dataset.dataset_source,
+                            # If more than one dataset is available go with the partial dataset.
+                            # If only one dataset is available just go with that one
+                            # Slack discussion: https://genomes-ebi.slack.com/archives/C010QF119N1/p1746094298003789
+                            # release=gd.dataset.genome_datasets[0],
+                            release=utils.fetch_proper_dataset(gd.dataset.genome_datasets),
```

Comment:
why is a list of genome datasets named `release` here?

Reply:
I myself, I'm confused, I promise I'll take a look at this and fix it it in another PR